CDCgov/prime-reportstream
# 5702 Pull Serializers out of BlobAccess

This PR takes the report serializing functionality out of BlobAccess.

Test Steps:
1. Verify all tests pass
2. Manually submit reports, and verify they are stored as blobs in your Azure Storage

## Changes
- Serialization logic that used to be on `BlobAccess` is now in the newly created class `ReportWriter`
- Added and updated test cases to improve coverage
- Updated references to BlobAccess where it was needed to maintain the same functionality.

## Linked Issues
- #5702
